core/src/main/java/io/micronaut/core/type/MutableHeaders.java

```diff
@@ -53,4 +53,19 @@ default MutableHeaders set(CharSequence header, CharSequence value) {
         add(header, value);
         return this;
     }
+
+    /**
+     * Sets an HTTP header if missing.
+     *
+     * @param header The header
+     * @param value  The value
+     * @return This headers
+     * @since 4.6
+     */
+    default MutableHeaders setIfMissing(CharSequence header, CharSequence value) {
```

Comment:
this calls `add` so should be named `addIfMissing`. Add nullability annotations to the arguments and null check the `header` argument